staging/src/k8s.io/client-go/INSTALL.md

```diff
@@ -143,6 +143,48 @@ After modifying, run `glide up -v` again to re-populate your /vendor directory.
 Optionally, Glide users can also use [`glide-vc`](https://github.com/sgotti/glide-vc)
 after running `glide up -v` to remove unused files from /vendor.
 
+### Go modules
+
+Dependency management tools are built into go 1.11+ in the form of [go modules](https://github.com/golang/go/wiki/Modules).
+
+If you are using go 1.11 or 1.12 and are working with a project located within `$GOPATH`,
+you must opt into using go modules:
+
+```sh
+export GO111MODULE=on
+```
+
+Ensure your project has a `go.mod` file defined at the root of your project.
+If you do not already have one, `go mod init` will create one for you:
+
+```sh
+go mod init
+```
+
+Indicate the matching versions of `k8s.io/client-go`, `k8s.io/api`, and `k8s.io/apimachinery` your project requires:
+
+```sh
+go get k8s.io/client-go@v10.0.0              # replace v10.0.0 with the required version (or use kubernetes-1.x.y tags if desired)
```

Comment:
All the module examples I've seen show this method of pulling client-go in, which I assume is due to the way client-go has mixed tagging. As far as I can tell there's no way to do otherwise - am I correct about this? If so, future compatibility with other dependencies is not guaranteed.

Would love to be wrong about this.

Reply:
this tells go what version of client-go to fetch

> which I assume is due to the way client-go has mixed tagging

not sure what you mean by "mixed tagging"

> If so, future compatibility with other dependencies is not guaranteed

The major versions indicate each release of client-go typically makes changes someone integrating with the go API has to react to. Not sure if that is what you meant.

Reply:
By mixed tagging, I mean both vx.0.0 and kubernetes-1.x.y. But I was wrong, sorry for the noise.